spi/core-spi/src/main/java/org/eclipse/dataspaceconnector/spi/system/ConfigurationExtension.java

```diff
@@ -27,4 +29,11 @@
     @Nullable
     String getSetting(String key);
 
+    /**
+     * Gets all configuration settings that start with a certain prefix (= filtering).
+     *
+     * @param prefix the prefix
+     * @return A map that contains all config entries that start with the prefix.
+     */
+    Map<String, String> getSettingsWithPrefix(String prefix);
```

Comment:
Why this `settingsWithPrefix` is needed?, should be enough to get all the settings on which to apply the `hierarchical` function. 
Filtering can be implemented by the client

Reply:
the reason for this is that we might not know the exact set of properties. This was developed with configuring JDBC drivers in mind, which can have different properties. 
With this, it is sufficient to state the prefix, which would be `edc.datasource.<DS_NAME>` and that'll give you all properties that are configured.

Reply:
what about an interface like:
```
Map<String, Object> getConfig(String path)
```
that will directly give an hierarcal structure of the configuration at that path so to permit the client to avoid creating hierarcy by itself? (as said in the main comment, the return type could be a `Config` class that wraps the contents and offer a better interface than `Map`)

Reply:
I assume you meant to type `Config getConfig(String path)`?

Reply:
Yes

Reply:
As I said, the client might not know about the structure, as this may be handled in a generic way. Then again, I might not have completely understood your idea :)

But since @jimmarino is waiting for this feature to get merged for his PR #541, which in turn blocks a few other features: would you be ok to merge this as-is and let's come up with a proposal that uses your `Config` variant?

Reply:
sure

Reply:
opened #557